src/main/java/com/google/cloud/spanner/pgadapter/utils/ClientAutoDetector.java

```diff
@@ -108,15 +122,16 @@ public ImmutableList<LocalStatement> getLocalStatements(ConnectionHandler connec
 
   /**
    * Returns the {@link WellKnownClient} that the detector thinks is connecting to PGAdapter based
-   * purely on the list of parameters.
+   * purely on the list of parameters. It will return UNSPECIFIED if no specific client could be
+   * determined.
    */
-  public static WellKnownClient detectClient(
+  public static @Nonnull WellKnownClient detectClient(
       List<String> orderParameterKeys, Map<String, String> parameters) {
     for (WellKnownClient client : WellKnownClient.values()) {
       if (client.isClient(orderParameterKeys, parameters)) {
         return client;
       }
     }
-    return null;
+    throw new IllegalStateException("UNSPECIFIED.isClient() should have returned true");
```

Comment:
Is this line reachable through any condition?

Reply:
No, it is not reachable. Added a comment to the code to clarify that.